2.evaluate_models/log_reg_plates_cp_fs_data/plate_4/nbconverted/evaluate_model.py

```diff
@@ -0,0 +1,169 @@
+plate_4_train_path = "1.train_models/log_reg_plates_cp_fs_data/plate_4"
+data_path = pathlib.Path(f"{root_dir}/{plate_4_train_path}/model_data")
+model_path = pathlib.Path(f"{root_dir}/{plate_4_train_path}/trained_models")
+split_path = data_path / "data_split_indices_log_reg_cp_fs_data_plate_4.tsv"
+le_path = data_path / "label_encoder_log_reg_cp_fs_data_plate_4.joblib"
+
+# Output paths
+fig_path = pathlib.Path("figures")
+fig_path.mkdir(parents=True, exist_ok=True)
+
+
+# ## Load models and model data
+
+# In[4]:
+
+
+# Load models
+logreg_model_shuf = load(model_path / model_shuf)
+logreg_model = load(model_path / model)
+
+# Load plate data
+platedf = pd.read_parquet(plate_path)
+
+# Load label encoder
+le = load(le_path)
+
+# Load split data indices as dataframe
+splitdf = pd.read_csv(split_path, sep="\t", index_col=0)
+
+
+# ## Merge the split data with the plate data
+
+# In[5]:
+
+
+platedf = pd.merge(platedf, splitdf, left_index=True, right_index=True)
+
+
+# ## Specify the feature column names
+
+# In[6]:
+
+
+feat_columns = logreg_model.feature_names_in_
+
+featdf = platedf[feat_columns]
+
+
+# ## Calculate Precision-Recall Curves
+
+# In[7]:
+
+
+data_split = defaultdict(list)
+split_column = "Metadata_split_name"
+
+for model, data_type in list(product((logreg_model, logreg_model_shuf), ("train", "test"))):
+
+    # Only use the shuffled training feature for calculating the shuffled model
+    # unshuffled test data is used to evaluate the shuffled model
+    if (model == logreg_model_shuf):
+        shuffled = "shuffled"
+        if (data_type == "train"):
+            X_split_name = f"X_{data_type}_shuf"
+        else:
+            X_split_name = "X_test"
+    else:
+        shuffled = "not_shuffled"
+        X_split_name = f"X_{data_type}"
+
+    # Split the data based on the comparison being made
+    Xdf = featdf.loc[platedf[split_column] == X_split_name]
+    ydf = platedf.loc[platedf[split_column] == f"y_{data_type}"]["Metadata_genotype"]
+    ydf = ydf.reindex(Xdf.index)
+
+    # Encode the genotype labels
+    y_val = le.transform(ydf)
+
+    # Predict probabilites from the X data
+    probas_pred = model.predict_proba(Xdf)
+
+    # Calculate the precision, recall data
+    precision, recall, proba_thresh = precision_recall_curve(y_val, probas_pred[:, -1])
+    proba_thresh = np.append(proba_thresh, np.nan)
+
+    # Calculate the suffled data
+    metric_len = precision.shape[0]
+
+    data_split["precision"].extend(precision.tolist())
+    data_split["recall"].extend(recall.tolist())
+    data_split["probability_threshold"].extend(proba_thresh.tolist())
+    data_split["model_type"].extend([shuffled] * metric_len)
+    data_split["split_type"].extend([data_type] * metric_len)
+
+# Convert dictionary of lists to a dataframe
+data_splitdf = pd.DataFrame(data_split)
+
+
+# # Plot precision-recall curves
+
+# In[8]:
+
+
+plt.figure(figsize=(20, 14))
```

Comment:
Consider making the plot size smaller. Smaller plot size makes smaller font appear larger.
Down sides to this is the graph looks "tighter" less spread out. 6,8 or 10,12 have worked well for me in the past.

Reply:
The difference was easy to see, thanks for the suggestion!